Libs/MRML/Core/vtkCodedEntry.cxx

```diff
@@ -0,0 +1,156 @@
+/*=auto=========================================================================
+
+Portions (c) Copyright 2017 Brigham and Women's Hospital (BWH) All Rights Reserved.
+
+See COPYRIGHT.txt
+or http://www.slicer.org/copyright/copyright.txt for details.
+
+=========================================================================auto=*/
+
+#include "vtkCodedEntry.h"
+
+// VTK include
+#include <vtkObjectFactory.h>
+
+// STD include
+#include <sstream>
+
+vtkStandardNewMacro(vtkCodedEntry);
+
+//----------------------------------------------------------------------------
+vtkCodedEntry::vtkCodedEntry()
+: CodingSchemeDesignator(NULL)
+, CodeValue(NULL)
+, CodeMeaning(NULL)
+{
+}
+
+//----------------------------------------------------------------------------
+vtkCodedEntry::~vtkCodedEntry()
+{
+  this->Initialize();
+}
+
+//----------------------------------------------------------------------------
+void vtkCodedEntry::Initialize()
+{
+  this->SetCodingSchemeDesignator(NULL);
+  this->SetCodeValue(NULL);
+  this->SetCodeMeaning(NULL);
+}
+
+;
+
+//----------------------------------------------------------------------------
+void vtkCodedEntry::SetSchemeValueMeaning(const std::string& scheme,
+  const std::string& value, const std::string& meaning)
+{
+  this->SetCodingSchemeDesignator(scheme.c_str());
+  this->SetCodeValue(value.c_str());
+  this->SetCodeMeaning(meaning.c_str());
+}
+
+//----------------------------------------------------------------------------
+void vtkCodedEntry::PrintSelf(ostream& os, vtkIndent indent)
+{
+  Superclass::PrintSelf(os,indent);
+  os << indent << "CodingSchemeDesignator: " << (this->CodingSchemeDesignator?this->CodingSchemeDesignator:"(none)") << "\n";
+  os << indent << "CodeValue: " << (this->CodeValue?this->CodeValue:"(none)") << "\n";
+  os << indent << "CodeMeaning: " << (this->CodeMeaning?this->CodeMeaning:"(none)") << "\n";
+}
+
+//----------------------------------------------------------------------------
+void vtkCodedEntry::Copy(vtkCodedEntry* aType)
+{
+  if (!aType)
+    {
+    return;
+    }
+  this->SetCodingSchemeDesignator(aType->GetCodingSchemeDesignator());
+  this->SetCodeValue(aType->GetCodeValue());
+  this->SetCodeMeaning(aType->GetCodeMeaning());
+}
+
+//----------------------------------------------------------------------------
+std::string vtkCodedEntry::GetAsPrintableString()
+{
+  std::string printable = std::string("(")
+    + (this->CodingSchemeDesignator ? this->CodingSchemeDesignator : "(none)") + ", "
+    + (this->CodeValue ? this->CodeValue : "(none)") + ", "
+    + (this->CodeMeaning ? this->CodeMeaning : "(none)") + ")";
```

Comment:
I agree that when there is comma then it may be more readable to have a different separator. However, adding double quotes all the time would make things a bit harder to read in general.

What about using a different separator?
Or just adding double quotes when there is comma in the text?